### PR TITLE
Configure babel to ignore babelrc and use babel-preset-yoshi

### PR DESCRIPTION
**Pull request by ronami**
### 🔦 Summary

This PR moves Babel's configuration from the developer to Yoshi: Yoshi will ignore `.babelrc` and `babel.config.js` and instead will pre-configure Babel with `babel-preset-yoshi`.

This change will allow us to configure Babel in the best way possible: Like transpiling server code to the current Node version.

### CODE REVIEW

File: packages/jest-yoshi-preset/package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
   },
   "dependencies": {
     "babel-jest": "^23.4.2",
+    "babel-preset-yoshi": "4.0.0",
```
Comment (ronami) [inline packages/jest-yoshi-preset/package.json]: move to dev deps

File: packages/yoshi-helpers/package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
   "license": "ISC",
   "dependencies": {
     "@babel/register": "^7.0.0",
+    "babel-preset-yoshi": "4.0.0",
```
Comment (ronami) [inline packages/yoshi-helpers/package.json]: move to dev deps

File: packages/yoshi/config/webpack.config.js [hunk 3]
```diff
@@ -415,6 +415,11 @@ function createCommonWebpackConfig({
             },
             {
               loader: 'babel-loader',
+              options: {
+                babelrc: false,
+                configFile: false,
+                presets: [require.resolve('babel-preset-yoshi')],
```
Comment (ronami) [inline packages/yoshi/config/webpack.config.js]: pass modules: false and remove `IN_WEBPACK`

File: packages/yoshi/src/commands/start.js [hunk 4]
```diff
@@ -262,24 +261,31 @@ module.exports = runner.command(
...
-          target: 'dist',
-          sourceMaps: true,
-        });
-        return appServer();
-      }
+            babelrc: false,
+            configFile: false,
+            presets: [[require.resolve('babel-preset-yoshi')]],
+          });
+          await appServer();
```
Comment (ronami) [inline packages/yoshi/src/commands/start.js]: use `return` to keep consistent

File: packages/yoshi/test/e2e.spec.js [hunk 5]
```diff
@@ -238,18 +227,13 @@ describe('Aggregator: e2e', () => {
     };
   }
 
-  function singleModuleWithJasmineAndES6Imports(runIndividualTranspiler) {
+  function singleModuleWithJasmineAndES6Imports() {
```
Comment (ronami) [inline packages/yoshi/test/e2e.spec.js]: inline it to its only use-case

File: packages/yoshi/test/jasmine.spec.js [hunk 6]
```diff
@@ -186,9 +185,7 @@ function jasmineSetup() {
 }
 
 function checkStdoutContains(test, str) {
-  return retryPromise(
-    { backoff: 100 },
-    () =>
-      test.stdout.indexOf(str) > -1 ? Promise.resolve() : Promise.reject(),
+  return retryPromise({ backoff: 100 }, () =>
+    test.stdout.indexOf(str) > -1 ? Promise.resolve() : Promise.reject(),
```
Comment (ronami) [inline packages/yoshi/test/jasmine.spec.js]: use `expect` instead

**Comment by ronami** [comment]
__Docs that we need to add:__

* Change the docs regarding using babel in general and explain how it behaves (server/client/testing ...)
* Today ESM directory (for libraries) require a specific configuration for `.babelrc`, need update.
* Remove `runIndividualTranspiler` from docs.

File: packages/yoshi-helpers/package.json [hunk 7]
```diff
@@ -23,16 +23,18 @@
     "server-destroy": "^1.0.1",
     "ts-node": "^3.3.0",
     "xmldoc": "^1.1.2",
-    "yoshi-config": "3.23.0"
+    "yoshi-config": "3.23.0",
+    "yoshi-helpers": "3.26.0"
```
Comment (ranyitz) [inline packages/yoshi-helpers/package.json]: redundant

**Pull request merged**